### DISCUSSION TITLE
Getting error "No public key encrypted session key packet found" when decrypting

### DISCUSSION THREAD

**Discussion by wo-yashsinghvi** [openpgpjs/openpgpjs#1646] [Q&A]
I am using the openpgpjs version 5.9.0 and getting the error while decrypting the data. Below is the code. Can someone please help me out with it?

--------------------------------------------------------------------------------------------------------------------------------------------------------

const openpgp =  require('openpgp');
const { readFileSync } = require('fs');

const fun3 = async () => {
    try {
        const passphrase = 'Admin@123';
        const privateKeyArmored = readFileSync('./private-key.asc', 'utf8');
        const privateKey = await openpgp.decryptKey({
            privateKey: await openpgp.readPrivateKey({ armoredKey: privateKeyArmored }),
            passphrase
        });

        const publicKeyArmored = readFileSync('./public-key.asc', 'utf8');
        const publicKey = await openpgp.readKey({ armoredKey: publicKeyArmored });

        const encryptedAndSignedMessage = readFileSync('./import.json.pgp', 'utf-8');
        var encryptedUint8 = new TextEncoder().encode(encryptedAndSignedMessage);
        const message = await openpgp.readMessage({
            binaryMessage: encryptedUint8
        });
        const { data } = await openpgp.decrypt({
            message,
            decryptionKeys: privateKey
        });
        console.log(data);
    } catch (e) {
        console.log('error', e);
    }
}

fun3();

--------------------------------------------------------------------------------------------------------------------------------------------------------

**Comment by larabr** [comment]
The error means that the message you are trying to decrypt is not encrypted using public keys -- it may be encrypted with a password instead. You can try decrypting with a different software to see if it works, or share the message here so we can check.

**Reply by wo-yashsinghvi** [reply]
@larabr Yes, I tried GUI-based software and I was able to decrypt it. I can't use software to decrypt the file I will need to place some code to decrypt it. Please suggest something.

Software link:- https://pgptool.github.io/

**Reply by wo-yashsinghvi** [reply]
@larabr Any update on this?

**Reply by twiss** [reply]
Hey :wave: Do you have an example message, and public key that it's encrypted with, so that we can take a look at it?

**Reply by twiss** [reply]
Actually -

> ```js
>     var encryptedUint8 = new TextEncoder().encode(encryptedAndSignedMessage);
>     const message = await openpgp.readMessage({
>         binaryMessage: encryptedUint8
>     });
> ```

this won't work. If `encryptedAndSignedMessage` is an armored message, you can do:

```js
    const message = await openpgp.readMessage({
        armoredMessage: encryptedAndSignedMessage
    });
```

otherwise, if it's a binary message, you need to convert it to a Uint8Array in a different way, depending on how the string is encoded. If it's a binary string, you can try something like

```js
    const message = await openpgp.readMessage({
        binaryMessage: Uint8Array.from(encryptedAndSignedMessage, str => str.charCodeAt(0))
    });
```
